openerp/api.py

```diff
@@ -182,6 +182,27 @@ def _onchange_partner(self):
     return lambda method: decorate(method, '_onchange', args)
 
 
+def onwrite(*args):
+    """ Trigger the decorated method when the chosen fields are changed.
+
+    Example::
+
+        @api.multi
+        @api.onwrite("partner_id", "user_id")
+        def _onwrite_partner_user_combine(self):
+            for s in self:
+                if s.partner_id.user_id != s.user_id:
+                    s.user_id = s.partner_id.user_id
+
+    .. warning::
+
+        ``@onwrite`` only supports simple field names, dotted names
+        (fields of relational fields e.g. ``partner_id.customer``) are not
+        supported and will be ignored.
```

Comment:
This PR is more a placeholder than an implementation. My idea was to make api provide such mechanism, whose lack usually makes people misuse constraints or write unnecessaryly duplicate code, and when they actually make constraints more strong (forbid editing and such), then take the time to make onwrite stronger (by enabling dotted syntax or, as you say, checking it).